be/src/http/action/stream_load.cpp

```diff
@@ -301,6 +301,14 @@ Status StreamLoadAction::_on_header(HttpRequest* http_req, std::shared_ptr<Strea
         }
     }
 
+    if (ctx->body_bytes == 0 && !ctx->is_chunked_transfer) {
```

Comment:
Ctx->body_bytes == 0 must be zero in on_header method, for it hasn't started receive data here yet, which begins in StreamLoadAction::_handle. I think you can check in  _handle or use http_req->header(HttpHeaders::CONTENT_LENGTH).empty() to check.